cc/transport/grpc_streaming_transport.h

```diff
@@ -25,10 +29,41 @@
 
 namespace oak::transport {
 
+template <class Service>
 class GrpcStreamingTransport : public EvidenceProvider, public Transport {
  public:
-  absl::StatusOr<::oak::session::v1::AttestationBundle> GetEvidence() override;
-  absl::StatusOr<std::string> Invoke(absl::string_view request_bytes) override;
+  explicit GrpcStreamingTransport(std::shared_ptr<grpc::Channel> channel)
+      : channel_stub_(Service::NewStub(channel)) {}
+
+  absl::StatusOr<::oak::session::v1::AttestationBundle> GetEvidence() override {
+    ::oak::session::v1::GetPublicKeyRequest request;
+    ::oak::session::v1::GetPublicKeyResponse response;
+    grpc::ClientContext context;
+
+    grpc::Status status = channel_stub_->GetPublicKey(&context, request, &response);
+    if (!status.ok()) {
+      return absl::InternalError("couldn't send GetPublicKeyRequest: " + status.error_message());
+    }
+
+    return response.attestation_bundle();
+  }
+
+  absl::StatusOr<std::string> Invoke(absl::string_view request_bytes) override {
+    ::oak::session::v1::InvokeRequest request;
+    request.set_encrypted_body(request_bytes);
+    ::oak::session::v1::InvokeResponse response;
+    grpc::ClientContext context;
+
+    grpc::Status status = channel_stub_->Invoke(&context, request, &response);
```

Comment:
I'm confused by this -- shouldn't it use the streaming method? In fact, could you remind me why we still have both the unary and the streaming protos?

Reply:
The class should hold onto the stub and use the Read() and Write() functions for streaming gRPCs.

Reply:
Updated to support streams

Reply:
> I'm confused by this -- shouldn't it use the streaming method? In fact, could you remind me why we still have both the unary and the streaming protos?

@ipetr0v can you answer this? Can / should we remove the unary protos?

Reply:
The PR has been updated to use streaming gRPC functions.

About the Unary protos, they will only be used once we implement the Attestation Verification Service. And only for one-shot requests.

So they could be removed for now.